packages/expo/src/Maps/MapView.web.tsx

```diff
@@ -51,14 +39,22 @@ class MapView extends React.Component<Props, State> {
   };
 
   render() {
-    const { onRegionChange, onPress, children } = this.props;
+    const { region, initialRegion = {}, onRegionChange, onPress, children } = this.props;
```

Comment:
My sense is the MapView API is designed to let it act as a controlled region) or uncontrolled component (initialRegion). If you re-render the MapView with a different initialRegion, it should be ignored. Might need to use a bit of state actually.